### DISCUSSION TITLE
adding texture to surfaces

### DISCUSSION THREAD

**Discussion by Marwa-AL** [cityjson/specs#179] [Q&A]
Hi all, 

Thank you for your efforts with CityJSON! I have a question regarding adding textures. 

I am currently calculating solar radiation incident on each wall of the building. The result would be values calculate at specified points as shown in the image below. I would like to save the values as texture of the building. Is that possible? if yes, how it can be done? 

![image](https://github.com/cityjson/specs/assets/66286563/9919f6e3-9c7e-4b59-bc90-f58bcecaff2e)

**Comment by hugoledoux** [comment]
There was an MSc thesis here at TUDelft that tackled this more or less, it's there:
http://resolver.tudelft.nl/uuid:0d6fe201-5676-48c0-850f-942d1ee9a353

The code related to it is there:
https://github.com/robinjo78/solarBAG

He discusses in Sect4.2.3 four options to store the results of solar potential analysis, but he didn't really do it in the end; except to summarise/average the values per surface, which he stores as semantics/attributes.

If you wanted to store the points, you could add one geometry of type `MultiPoint` and attach semantics/attributes to each point (it's possible, see https://www.cityjson.org/specs/2.0.0/#semantics-of-geometric-primitives). 

The best option is, I guess, to create an [Extension](https://www.cityjson.org/extensions/) and extend the Building type to add one property `solar-potential` and document how you want the information to be structured.
